lib/workload/stateful/stacks/access-key-secret/index.ts

```diff
@@ -0,0 +1,61 @@
+import { SecretValue, Stack, StackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { AccessKey, AccessKeyStatus, PolicyStatement, User } from 'aws-cdk-lib/aws-iam';
+import { ISecret, Secret } from 'aws-cdk-lib/aws-secretsmanager';
+
+/**
+ * Props for the user access key stack.
+ */
+export type AccessKeySecretStackProps = {
+  /**
+   * Name of the secret to store the access tokens.
+   */
+  secretName: string;
+  /**
+   * The name of the user.
+   */
+  userName: string;
+  /**
+   * The policies to add to the user.
+   */
+  policies: PolicyStatement[];
+};
+
+/**
+ * This component creates long-lived user access tokens inside secrets manager. This is used to authorize actions like
+ * creating long-lived pre-signed URLs that can last up to 7 days.
+ */
+export class AccessKeySecret extends Stack {
```

Comment:
Perhaps could be useful as stand-alone component, i.e. managed IAM user with key rotation and key access through Secrets Manager... (for future consideration only)